.github/workflows/python_unit_test.yaml

```diff
@@ -1,49 +1,66 @@
 name: Run Python unit tests using Poetry, with a Postgres backend
 
-on: 
+on:
   workflow_call:
     inputs:
       setup-script-location:
         description: 'A URL identifying a Bash script that can run commands needed for service-specific test setup'
-        required: true
+        required: false
         type: string
-    
-env: 
+      python-version:
+        description: 'The Python version to use and defaults to latest'
+        required: false
+        default: 'latest'
+        type: string
+      postgres-needed:
+        description: 'Whether or not a Postgres backend is needed'
+        required: false
+        default: 'false'
+        type: boolean
+
+env:
   PGPASSWORD: postgres
 
 jobs:
   run_tests:
     runs-on: ubuntu-latest
-    
-    #Defining the Postgres DB service we will need to run the unit tests
+
+    #Defining the Postgres DB service we will need to run the unit tests if postgres-needed is true
     services:
+      if: ${{ inputs.postgres-needed }}
       postgres:
         image: postgres:9.6-bullseye
         ports:
           - "5432:5432"
         env:
-          POSTGRES_PASSWORD: postgres
+          POSTGRES_PASSWORD: ${{ secrets.default_postgres_password }}
         options: >-
           --health-cmd pg_isready
           --health-interval 10s
           --health-timeout 5s
           --health-retries 5
-    
+
     steps:
       - name: Check out code
         uses: actions/checkout@v3
```

Comment:
I think you need to add `persist-credentials:  false` it defaults to true. https://github.com/actions/checkout#usage